src/main/java/io/github/syst3ms/skriptparser/expressions/ExprStringCharAt.java

```diff
@@ -0,0 +1,54 @@
+package io.github.syst3ms.skriptparser.expressions;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import io.github.syst3ms.skriptparser.Parser;
+import io.github.syst3ms.skriptparser.lang.Expression;
+import io.github.syst3ms.skriptparser.lang.TriggerContext;
+import io.github.syst3ms.skriptparser.parsing.ParseContext;
+import io.github.syst3ms.skriptparser.util.DoubleOptional;
+
+/**
+ * The character at a given position in a string. Note that indices in Skript start at 1.
+ *
+ * @name Character at
+ * @pattern char[acter] at [(index|pos[ition])] %number% (of|in) %string%
+ * @since ALPHA
+ * @author Olyno
+ */
+public class ExprStringCharAt implements Expression<String> {
+
+	static {
+		Parser.getMainRegistration().addExpression(
+			ExprStringCharAt.class,
+			String.class,
+			true,
+			"char[acter] at [(index|pos[ition])] %integer% (of|in) %string%"
+		);
+	}
+
+	private Expression<Integer> position;
+	private Expression<String> value;
+
+	@SuppressWarnings("unchecked")
+	@Override
+	public boolean init(final Expression<?> @NotNull [] expressions, final int matchedPattern, @NotNull final ParseContext parseContext) {
+		position = (Expression<Integer>) expressions[0];
+		value = (Expression<String>) expressions[1];
+		return true;
+	}
+
+	@Override
+	public String[] getValues(final TriggerContext ctx) {
```

Comment:
Why are these all final? Maybe you have some kind of option enabled in your IDE, but these should not be final (applies to all methods).
Also make sure to look if these requests (removing finals and NotNull annotations) apply to your other PRs as well.